### ISSUE TITLE
Visual Studio runtime assert when iterating parsed options

### ISSUE THREAD

**Issue by acronce** [jarro2783/cxxopts#362]
If you implement iteration of parsed options (like in the example code) and build under Visual Studio, there will be a Debug runtime assert when attempting to access the first key/value pair.

The assert message is `Expression: vector iterators incompatible`

We verified this problem under VS2017 and VS2022. It does not happen with Xcode 13 (Apple clang version 13.1.6) or clang version 10.0.0-4ubuntu under Linux.

You can reproduce this issue by building and running the `example` target from the latest master (git hash ab1dc0f at the time of this writing). The assert happens on line number 93 of `example.cpp`.

Thanks in advance for any help with this problem.

**Comment by acronce** [comment]
Here's a screenshot of the assert:

<img width="364" alt="Screen Shot 2022-08-25 at 4 49 12 PM" src="https://user-images.githubusercontent.com/61846960/186791975-f88c1647-4a19-4055-9d95-c36a04c98c7d.png">

**Comment by jarro2783** [comment]
I think I know what's going on, it seems like visual studio's standard library does an extra check that two iterators point to the same container. I'll try to come up with a fix.

**Comment by acronce** [comment]
Checking in to see if there's any news on this issue. Thanks in advance.

**Comment by jarro2783** [comment]
Not yet, I'll try to get to it soon.

**Comment by acronce** [comment]
Any word on this issue?

Note that it's not critical because we've worked around it for now. But it would be nice to see it fixed properly.

Thanks in advance for your help.

**Comment by jarro2783** [comment]
Not yet sorry. I'll try to get to it soon.

**Comment by jarro2783** [comment]
Can you try the above commit to check if it is fixed?

**Comment by acronce** [comment]
Thanks for trying to fix this. Unfortunately I'm still seeing problems.

I can't reproduce the original problem because the example returns an error:

```
error parsing options: Argument '' failed to parse
```

This is new behavior, so I can't tell if the original problem is fixed or not.

After that I tried running `options_test.exe`, which fails with a similar assert, originating from line number 1594 of `cxxopts.hpp`.

<img width="403" alt="Screenshot 2023-01-15 at 5 33 34 PM" src="https://user-images.githubusercontent.com/61846960/212580571-b929483e-9783-49e2-9cd9-a4137ce165d7.png">

The branch that I checked out was "fix_iter_access" (commit 65b4e91f). Hopefully that's the correct branch.
